gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GqlQuery.java

```diff
@@ -379,9 +379,11 @@ protected void populatePb(com.google.datastore.v1beta3.RunQueryRequest.Builder r
   }
 
   @Override
-  protected GqlQuery<V> nextQuery(com.google.datastore.v1beta3.QueryResultBatch responsePb) {
-    // See issue #17
-    throw new UnsupportedOperationException("paging for this query is not implemented yet");
+  protected StructuredQuery<V> nextQuery(com.google.datastore.v1beta3.RunQueryResponse responsePb) {
+    return new StructuredQuery.Builder<>(type())
```

Comment:
namespace is not handled correctly.
Also, I think you can just do the following and without any change in StructuredQuery (other than the method param type, using RunQueryResponse):
```java
return StructuredQuery.fromPb(type(), namespace(), responsePb.getQuery()).nextQuery(responsePb);
```